cirq/contrib/qasm_import/qasm.py

```diff
@@ -17,10 +17,7 @@
 
 
 class QasmCircuitParser:
-    """QasmCircuitParser is currently partially developed, not functional,
-    DO NOT USE.
-    TODO(https://github.com/quantumlib/Cirq/issues/1548)
-    It will serve as the entrypoint for parsing QASM files."""
+    """Provides parsing of OpenQASM files."""
```

Comment:
Is this just parsing qasm files or qasm strings too? The example on here provides a string as input.

Reply:
What's the difference between `QasmParser` and `QasmCircuitParser`? Based on this and `QasmParser`'s docstring, this is for files and `QasmParser` is for strings. Could we combine them so a user can pass a variable declaring which format his/her qasm is in?

Reply:
It looks like `QasmParser` is the YACC parser for the QASM grammar and shouldn't be used directly.  Maybe `QasmCircuitParser` should be renamed to `QasmCircuitImporter` for clarity.

Reply:
Hmm...this docstring is confusing - we are just parsing strings, not files. 
Yes, the `QasmParser` is the YACC parser. 
`QasmCircuitParser` delegates a subset of `QasmParser`'s results that is relevant to the user - i.e the parsed circuit.